boa/src/builtins/value/mod.rs

```diff
@@ -614,7 +614,7 @@ impl ValueData {
         let val = val.into();
 
         if let Self::Object(ref obj) = *self {
-            if let ObjectData::Array = obj.borrow().data {
+            if obj.borrow().data.is_array() {
```

Comment:
I would add this function also to the `Object`, in order to not need to do `obj.data.is_array()` and just do `obj.is_array()`. They can be aliases, though.